swap_meet/vendor.py

```diff
@@ -1,2 +1,75 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = [] if inventory is None else inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.remove(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            other_vendor.add(my_item)
+            return True
+        return False
+
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+
+        result = self.swap_items(other_vendor, my_first_item, their_first_item)
+        return result
+
+    def get_by_category(self, category):
+        category_items_list = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_items_list.append(item)
+        return category_items_list
+
+    def get_best_by_category(self, category):
+        category_items_list = self.get_by_category(category)
+        if category_items_list:
```

Comment:
How could we change this into a guard statement that lets us unindent the intended flow of the function?

Reply:
`    def get_best_by_category(self, category):
        category_items_list = self.get_by_category(category)
        if not category_items_list:
            return None

        best_item = max(category_items_list,
                        key=lambda item: item.condition)
        return best_item`